[Motions 2023 11 cwg 1] P3046R0 Core Language Working Group "ready" Issues (#6684)
cplusplus/draft: source/expressions.tex

Its operand shall be a prvalue of type ``pointer to \tcode{T}'',
where \tcode{T} is an object or function type.

> The operator yields an lvalue of type \tcode{T}

Reviewer: ```suggestion
The operator yields an lvalue of type \tcode{T}.
```

\rSec3[expr.unary.noexcept]{\tcode{noexcept} operator}

\indextext{\idxcode{noexcept}}%
\indextext{expression!\idxcode{noexcept}}%

\begin{bnf}
\nontermdef{noexcept-expression}\br
  \keyword{noexcept} \terminal{(} expression \terminal{)}
\end{bnf}

\pnum
The operand of the \keyword{noexcept} operator

> is an unevaluated operand\iref{expr.context}.

Reviewer: ```suggestion
is an unevaluated operand\iref{term.unevaluated.operand}.
```

Author: Added in separate commit.

Otherwise, the result of each of the operators is unspecified.
\begin{note}
A relational operator applied
to unequal function pointers
yields an unspecified result.

> A pointer value of type ``pointer to \cv void''

Reviewer: ![1700157938](https://github.com/cplusplus/draft/assets/21071787/4437fa14-5e83-4cd5-a83b-93ca7acbe667)
```suggestion
A pointer value of type ``pointer to \cv{}~\keyword{void}''
```

Author: thanks!
